### PR TITLE
added consumers to the documentation

### PR DESCRIPTION
**Pull request by Taliik**
## Summary of the changes / Why this is an improvement
Added documentation for the following commands:
- `croud consumers deploy`
- `croud consumers list`
- `croud consumers edit`

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by Taliik** [comment]
I wanna put 2 things up for discussion:

- use a --config argument
We could use a --config argument instead of --consumer-connection-string, …, because this adds a lot of arguments which we will need individually for each consumer type, because we would need to name them —eventhub-consumer-connection-string instead if we would not do a config. This gives us more flexibility and makes it easier to add new types of consumers without needing to add more arguments each time.

- Help texts for arguments
We could add a sphinx extension that would allow us to for example have a `helpfor::` in our documentation where we could specify the help for a given subcommand, e.g. `croud clusters deploy --help`. This would be a nice to have since we are reusing arguments for different commands with different purpose, and we cannot define all those in the help text of the argument directly.

What do you guys think about this?

File: docs/docs/commands.txt [hunk 1]
```diff
@@ -414,6 +414,146 @@ Available options:
...
+| ``--consumer-schema <STRING>``      | Yes      | The name of the schema in which to       |
+|                                     |          | insert.                                  |
++-------------------------------------+----------+------------------------------------------+
+| ``--project-id <STRING>``           | Yes      | The identifier of the project in which   |
+|                                     |          | the consumer should be deployed.         |
++-------------------------------------+----------+------------------------------------------+
+| ``--product-name <STRING>``         | Yes      | The name of the desired consumer         |
+|                                     |          | product.                                 |
++-------------------------------------+----------+------------------------------------------+
+| ``--tier <STRING>``                 | Yes      | The tier of the consumer.                |
```
[the remark this answers is not in the document]
Comment (chaudum) [reply docs/docs/commands.txt]: That is not correct. We drop the scale units in favour of an argument specifying the number of desired instances. Tiers still exist as they define the resources for the consumers.

[the remark this answers is not in the document]
Comment (chaudum) [reply docs/docs/commands.txt]: `--instances` or `--num-instances` is OK for me

[the remark this answers is not in the document]
Comment (MarkusH) [reply docs/docs/commands.txt]: > That is not correct. We drop the scale units in favour of an argument specifying the number of desired instances. Tiers still exist as they define the resources for the consumers.

:man_facepalming:

**Comment by chaudum** [comment]
> I wanna put 2 things up for discussion:
> 
>     * use a --config argument
>       We could use a --config argument instead of --consumer-connection-string, …, because this adds a lot of arguments which we will need individually for each consumer type, because we would need to name them —eventhub-consumer-connection-string instead if we would not do a config. This gives us more flexibility and makes it easier to add new types of consumers without needing to add more arguments each time.

Yes please :+1: 

>     * Help texts for arguments
>       We could add a sphinx extension that would allow us to for example have a `helpfor::` in our documentation where we could specify the help for a given subcommand, e.g. `croud clusters deploy --help`. This would be a nice to have since we are reusing arguments for different commands with different purpose, and we cannot define all those in the help text of the argument directly.

Yes please :+1:  Also, any other idea how to keep documentation and code in sync is welcome!

File: docs/docs/commands.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Taliik) [reply docs/docs/commands.txt]: Ok, so we are dropping `--unit` in favour of `--num-instances`?

File: docs/docs/commands.txt [hunk 2]
```diff
@@ -414,6 +414,146 @@ Available options:
...
+|                                     |          | - ``XS``                                 |
+|                                     |          | - ``S``                                  |
+|                                     |          | - ``M``                                  |
+|                                     |          | - ``L``                                  |
+|                                     |          | - ``XL``                                 |
++-------------------------------------+----------+------------------------------------------+
+| ``--unit <INT>``                    | No       | The scale unit of the deployed consumer  |
+|                                     |          | product.                                 |
++-------------------------------------+----------+------------------------------------------+
+| ``--consumer-connection-string      | No       | The connection string of the consumer.   |
```
Comment (m-kharbat) [inline docs/docs/commands.txt]: do we want to keep `*-connection-string` ? can we go with an abbreviation such as `*-conn-str`?

Comment (Taliik) [reply docs/docs/commands.txt]: i'm fine with that, makes the commands shorter.

**Comment by MarkusH** [comment]
> > I wanna put 2 things up for discussion:
> > ```
> > * use a --config argument
> >   We could use a --config argument instead of --consumer-connection-string, …, because this adds a lot of arguments which we will need individually for each consumer type, because we would need to name them —eventhub-consumer-connection-string instead if we would not do a config. This gives us more flexibility and makes it easier to add new types of consumers without needing to add more arguments each time.
> > ```
> 
> Yes please +1

What would that look like? You mean loading the config from a a file? That seems to add a huge overhead to me. I'd need to change some config file each time I need to create a consumer. Which is unlikely going to be that regularly.

**Comment by chaudum** [comment]
> > > I wanna put 2 things up for discussion:
> > > ```
> > > * use a --config argument
> > >   We could use a --config argument instead of --consumer-connection-string, …, because this adds a lot of arguments which we will need individually for each consumer type, because we would need to name them —eventhub-consumer-connection-string instead if we would not do a config. This gives us more flexibility and makes it easier to add new types of consumers without needing to add more arguments each time.
> > > ```
> > 
> > 
> > Yes please +1
> 
> What would that look like? You mean loading the config from a a file? That seems to add a huge overhead to me. I'd need to change some config file each time I need to create a consumer. Which is unlikely going to be that regularly.

I could imagine it as a mutually exclusive group to the other config arguments. The `--config=FILE` argument could be fed by a JSON or YAML file which contains the keys.
Maybe for only 3 specific arguments it is not so much of a help, but if you have a lot of arguments. I could see it being very handy.

File: docs/docs/commands.txt [hunk 3]
```diff
@@ -414,6 +414,146 @@ Available options:
...
+| ``--consumer-table <STRING>``       | No       | The name of the table in which to        |
+|                                     |          | insert.                                  |
++-------------------------------------+----------+------------------------------------------+
+| ``--consumer-schema <STRING>``      | No       | The name of the schema in which to       |
+|                                     |          | insert.                                  |
++-------------------------------------+----------+------------------------------------------+
+| ``--consumer-group <STRING>``       | No       | The consumer group from which to consume |
+|                                     |          | from.                                    |
++-------------------------------------+----------+------------------------------------------+
+| ``--consumer-conn-str <STRING>``    | No       | The connection string of the consumer.   |
```
Comment (chaudum) [inline docs/docs/commands.txt]: If we want to make them shorter, I'd prefer using `--consumer-dsn` rather than `--consumer-conn-str`.

**Pull request merged**